src/app/utils/chart-utils.ts

```diff
@@ -69,3 +69,38 @@ export const filterHourlyActiveAccounts = (
 ): ActiveAccounts[] | undefined => {
   return windows?.filter((value, index) => index % 12 === 0)
 }
+
+export const sumBucketsByChucks = <T>(
+  buckets: T[] | undefined,
+  sumKey: keyof T,
+  dateKey: keyof T,
+  numOfChunks: number,
+) => {
+  if (!buckets) {
+    return []
+  }
+
+  const chunkArray = buckets.reduce(
+    (accArray, item, index) => {
+      const chunkIndex = Math.floor(index / numOfChunks)
+
+      if (!accArray[chunkIndex]) {
+        accArray[chunkIndex] = []
+      }
+
+      accArray[chunkIndex].push(item)
+
+      return accArray
+    },
+    [[] as T[]],
+  )
+
+  return chunkArray
+    .map(groupedBy => {
+      return {
+        [sumKey]: groupedBy.reduce((acc, curr) => acc + (curr[sumKey] as number), 0),
+        [dateKey]: groupedBy[0][dateKey],
+      }
+    })
+    .flat()
```

Comment:
Is there a need to call flat method at this point ?

Reply:
Good point, removed.